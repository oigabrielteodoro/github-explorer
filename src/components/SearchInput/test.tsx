import React from 'react'
import { render, screen, fireEvent } from '@testing-library/react'

import { SearchInput } from '.'

const mockedOnChange = jest.fn()

const props = {
  placeholder: 'Digite aqui',
  onChange: mockedOnChange,
}

describe('<SearchInput />', () => {
  it('should render correctly', () => {
    const { container } = render(<SearchInput {...props} />)

    expect(screen.getByPlaceholderText(/digite aqui/i)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should change the value when onChange is called', () => {
    render(<SearchInput {...props} />)

    const input = screen.getByPlaceholderText(/digite aqui/i)

    expect(input).toBeInTheDocument()

    fireEvent.change(input, { target: { value: 'John Doe' } })

    expect(mockedOnChange).toHaveBeenCalled()
  })
})
