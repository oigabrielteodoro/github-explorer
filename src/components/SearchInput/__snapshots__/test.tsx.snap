// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchInput /> should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-right: auto;
  background: #fff;
  border-radius: 0.5rem;
}

.c0 button {
  border: 0;
  color: #fff;
  padding: 2.4rem 3rem;
  background: #04D361;
  font-size: 2rem;
  font-weight: 500;
  border-radius: 0 0.5rem 0.5rem 0;
  -webkit-transition: 0.3s ease-in-out;
  transition: 0.3s ease-in-out;
  min-width: 21rem;
}

.c0 button:hover {
  -webkit-filter: brightness(0.9);
  filter: brightness(0.9);
}

.c0 input {
  border: 0;
  background: transparent;
  color: #3a3a3a;
  font-size: 2rem;
  padding: 2.4rem 3rem;
  border-radius: 0.5rem 0 0 0.5rem;
  width: 50rem;
}

.c0 input::-webkit-input-placeholder {
  color: #a8a8b3;
}

.c0 input::-moz-placeholder {
  color: #a8a8b3;
}

.c0 input:-ms-input-placeholder {
  color: #a8a8b3;
}

.c0 input::placeholder {
  color: #a8a8b3;
}

<div
  class="c0"
>
  <input
    placeholder="Digite aqui"
  />
  <button
    type="button"
  >
    Pesquisar
  </button>
</div>
`;
